source/posix/socket.c

```diff
@@ -493,15 +504,6 @@ int aws_socket_connect(
                 socket->event_loop = NULL;
                 goto err_clean_up;
             }
-
-            uint64_t timeout = 0;
-            aws_event_loop_current_clock_time(event_loop, &timeout);
-
-            timeout += aws_timestamp_convert(
-                socket->options.connect_timeout_ms, AWS_TIMESTAMP_MILLIS, AWS_TIMESTAMP_NANOS, NULL);
-            /* schedule a task to run at the connect timeout interval, if this task runs before the connect
-             * happens, we consider that a timeout. */
-            aws_event_loop_schedule_task_future(event_loop, &socket_impl->connect_args->task, timeout);
```

Comment:
There's still a weird race in the stupid world where someone sets a VERY LOW timeout value. The timeout task could execute while this function is still running.

What if we kept the old order of things, but cached the task pointer:
1) `struct aws_task *timeout_task = &socket_impl->connect_args->task;`
2) then subscribed to io events
3) then scheduled `timeout_task`